main.tf

```diff
@@ -445,7 +445,8 @@ resource "azurerm_kubernetes_cluster" "main" {
     for_each = var.log_analytics_workspace_enabled ? ["oms_agent"] : []
```

Comment:
I'll try to fix

Reply:
I actually think this works okay already? I don't understand why `msi_auth_for_monitoring_enabled` != `null` should create `oms_agent` block. 

1. If `log_analytics_workspace_enabled` = **true**, and `msi_auth_for_monitoring_enabled` = **null**: 
`oms_agent` block is OK

2. If `log_analytics_workspace_enabled` = **true**, and `msi_auth_for_monitoring_enabled` = **True || False**:  
`oms_agent` block is OK

3. If `log_analytics_workspace_enabled` = `false`: 
`oms_agent` block is SKIPPED.

Reply:
Let me know if there's something I'm missing

Reply:
@lonegunmanb I believe @admincasper  is right. 

The value `msi_auth_for_monitoring_enabled` is about enabling managed identity authentication for the oms agent to ship data to the log analytics workspace.

The following statement is wrong:

>> now if var.log_analytics_workspace_enabled is true OR var.msi_auth_for_monitoring_enabled is not null should create an oms_agent block. 

if `var.log_analytics_workspace_enabled` is `false` the block should not be created. The value `msi_auth_for_monitoring_enabled` does not make sense if `log_analytics_workspace_enabled` is `false`.

Do you think we need instead of `precondition` that prevents `msi_auth_for_monitoring_enabled != null` when `log_analytics_workspace_enabled != true` ?